### PR TITLE
docs(python): note ordering guarantee for groupby

### PR DESCRIPTION
**Pull request by MarcoGorelli**
would've closed https://github.com/pola-rs/polars/issues/9754 if it was still open

### CODE REVIEW

File: py-polars/polars/dataframe/frame.py [hunk 1]
```diff
@@ -4641,6 +4641,16 @@ def groupby(
             Settings this to ``True`` blocks the possibility
             to run on the streaming engine.
 
+            .. note::
+
+                Within each group, the order of rows is always preserved, regardless
+                of this argument.
```
Comment (stinodego) [inline py-polars/polars/dataframe/frame.py:4647]: It's a bit more readable without the extra blank line, in my opinion.

```suggestion
            .. note::
                Within each group, the order of rows is always preserved, regardless
                of this argument.
```

**Review by stinodego — APPROVED** [review]
Thanks, good to have this in there. Added one minor suggestion, apply the commit if you please (or don't).

File: py-polars/polars/dataframe/frame.py (hunk 1 above)
Comment (MarcoGorelli) [reply py-polars/polars/dataframe/frame.py]: sure, thanks! I don't think it makes a difference when the docs are rendered, but still, good to have one less line of code

Comment (stinodego) [reply py-polars/polars/dataframe/frame.py]: Exactly, thanks!

**Pull request merged**